### DISCUSSION TITLE
what are families?

### DISCUSSION THREAD

**Discussion by itaysk** [godbout/kindaVim.blahblah#212] [Q&A]
picking up the discussion from https://github.com/godbout/kindaVim.docs/issues/139#issuecomment-1829766244 (don't want to pollute that issue further).

I'm trying to clarify what are families and how they work? I didn't find anything on this in the [docs](https://github.com/godbout/kindaVim.docs). The only vague reference to families are in the app settings:

> kindaVim handles many scenarios automatically, but some apps may require manual categorization into the Families below. This is because those apps may be buggy, restricted, returning improper Accessibility data, or you just want kindaVim to ignore them. If an app is recognized by The Wizard, you'll get a summary of the issues) found in that app, and suggestions about which Family to put it in according to your tastes and needs. To use The Wizard, drop apps into the placeholder on the right.

But that doesn't explain much about what is a Family and what effect the families below have. The text under each family explains what makes classifies the family but now what effect it has.

Also, I'm not sure what's the relationship between the 4 families (PG-R, Electron, Key Mapping, 911) and the 2 strategies (Keyboard, Accessibility). 

2 examples:

Yesterday I tried to use kV in Safari. It generally worked but then I tried to use `~` which didn't and I didn't understand why. I tried to drag safari to the wizard but it didn't work (later I found that asking the wizard from the menubar works). I found GH Issue where [you suggested it should be in PG-R](https://github.com/godbout/kindaVim.docs/issues/103#issuecomment-1200333697) so I added it, but I'm not sure why or what it did.

Today I tried to use kV in Word. it generally worked but `w` didn't. So I asked the wizard, and it said "Microsoft apps are mostly doing their own thing ... so that goes straight in the 911 Family". The 911 family says it's "For apps that don't even expose whether we're dealing with text nor not. Last resort.". Again, this doesn't explain to me what affect or implications 911 has.

I think a new markdown page explaining what are families, how they are related to strategies, what considerations to make when choosing families will be helpful.

Side note - the tabbed UI is also a little confusing since I need to switch between Auto and Keyboard. At first I thought this is another kind of configurations, meaning I need to choose if I want to be on auto or on keyboard.

**Comment by godbout** [comment]
this is gonna be my first quick answer, to try to clarify both for your side and from mine, so that i can help better later.

> I'm trying to clarify what are families and how they work? I didn't find anything on this in the [docs](https://github.com/godbout/kindaVim.docs).

the most important thing is actually `Strategies`, that are then split into `Families`. the strategies are talked about here: https://github.com/godbout/kindaVim.docs#current-vim-implementation and each of them have a bit of blah blah before showing which moves are implemented. ([here](https://github.com/godbout/kindaVim.docs/blob/master/AccessibilityStrategyImplementation.md#accessibility-strategy) and [here](https://github.com/godbout/kindaVim.docs/blob/master/KeyboardStrategyImplementation.md#keyboard-strategy))

somehow it was deliberate to not put that much info in the README, coz people don't read README and go directly using the apps. so there's stuff in the app, and if i'm not wrong, more than you've seen. see below.

The only vague reference to families are in the app settings:
> 
> > kindaVim handles many scenarios automatically, but some apps may require manual categorization into the Families below. This is because those apps may be buggy, restricted, returning improper Accessibility data, or you just want kindaVim to ignore them. If an app is recognized by The Wizard, you'll get a summary of the issues) found in that app, and suggestions about which Family to put it in according to your tastes and needs. To use The Wizard, drop apps into the placeholder on the right.
> 
> But that doesn't explain much about what is a Family and what effect the families below have. The text under each family explains what makes classifies the family but now what effect it has.
> 
> Also, I'm not sure what's the relationship between the 4 families (PG-R, Electron, Key Mapping, 911) and the 2 strategies (Keyboard, Accessibility).

the stuff is explained here:

https://github.com/godbout/kindaVim.docs/assets/121373/5df825b3-10ad-440c-8bc5-11a29be871db

> 
> 2 examples:
> 
> Yesterday I tried to use kV in Safari. It generally worked but then I tried to use `~` which didn't and I didn't understand why. I tried to drag safari to the wizard but it didn't work (later I found that asking the wizard from the menubar works).

dragging didn't work? should work fine. you need to drag an app to the icon in The Wizard.

https://github.com/godbout/kindaVim.docs/assets/121373/3f192694-2290-415d-93ed-c98c72c2c461

> I found GH Issue where [you suggested it should be in PG-R](https://github.com/godbout/kindaVim.docs/issues/103#issuecomment-1200333697) so I added it, but I'm not sure why or what it did.

when you drop an app, it explains you if there's an issue, and you have buttons that appear to add the app for you.

https://github.com/godbout/kindaVim.docs/assets/121373/86d8e64e-355e-4477-9f62-e294d5f3fb5f

> Today I tried to use kV in Word. it generally worked but `w` didn't. So I asked the wizard, and it said "Microsoft apps are mostly doing their own thing ... so that goes straight in the 911 Family". The 911 family says it's "For apps that don't even expose whether we're dealing with text nor not. Last resort.". Again, this doesn't explain to me what affect or implications 911 has.

so can you check the explanations that show up when you go over the information icon, as shown above? and see if it helps clarifying? and come back to me. thanks.

> I think a new markdown page explaining what are families, how they are related to strategies, what considerations to make when choosing families will be helpful.
> 
> Side note - the tabbed UI is also a little confusing since I need to switch between Auto and Keyboard. At first I thought this is another kind of configurations, meaning I need to choose if I want to be on auto or on keyboard.

i see. thanks. yeah i wasn't that happy about it but you can't do tabs within tabs with SwiftUI lol. so i had to find some other way. might need a refresh. thanks again.
